### ISSUE TITLE
If a texture has a `#` in the asset name, it cannot be downloaded from the editor or rendered in the view

### ISSUE THREAD

**Issue by yaustar** [playcanvas/editor#814]
Repro: Add a # to any texture asset name. See that it disappears from the preview and cannot be downloaded

https://user-images.githubusercontent.com/16639049/177303690-87960d78-e86c-4b04-bc43-f715387b7b0e.mp4

**Comment by yak32** [comment]
not CloudFront related - it doesn't work even locally

**Comment by willeastcott** [comment]
A `#` in a URL is link to a subsection on the page. For example:

https://developer.playcanvas.com/en/user-manual/introduction/#key-features

Presumably the `#` should be encoded.

**Comment by yak32** [comment]
Yeah I spent some time investigating it and it seems we don't encode it properly in some place

**Comment by yak32** [comment]
https://github.com/playcanvas/editor-api/pull/60

**Comment by yaustar** [comment]
Does this fix work on published builds as well?

**Comment by yak32** [comment]
published builds are not broken, it's just the editor issue

**Comment by yaustar** [comment]
This also applies to folder names 

https://playcanvas.com/project/1044803/overview/folder-name-bug

<img width="651" alt="image" src="https://user-images.githubusercontent.com/16639049/221995213-a0f6f348-928f-4fe6-abea-40e79cd6d6c8.png">
